src/TorchSharp/TorchVision/models/ResNet.cs

```diff
@@ -0,0 +1,352 @@
+                return new ResNet(
+                    "ResNet34",
+                    (in_planes, planes, stride) => new BasicBlock(in_planes, planes, stride),
+                    BasicBlock.expansion, new int[] { 3, 4, 6, 3 },
+                    numClasses,
+                    device);
+            }
+
+            public static ResNet ResNet50(int numClasses, Device device = null)
+            {
+                return new ResNet(
+                    "ResNet50",
+                    (in_planes, planes, stride) => new Bottleneck(in_planes, planes, stride),
+                    Bottleneck.expansion, new int[] { 3, 4, 6, 3 },
+                    numClasses,
+                    device);
+            }
+
+            public static ResNet ResNet101(int numClasses, Device device = null)
+            {
+                return new ResNet(
+                    "ResNet101",
+                    (in_planes, planes, stride) => new Bottleneck(in_planes, planes, stride),
+                    Bottleneck.expansion, new int[] { 3, 4, 23, 3 },
+                    numClasses,
+                    device);
+            }
+
+            public static ResNet ResNet152(int numClasses, Device device = null)
+            {
+                return new ResNet(
+                    "ResNet101",
+                    (in_planes, planes, stride) => new Bottleneck(in_planes, planes, stride),
+                    Bottleneck.expansion, new int[] { 3, 4, 36, 3 },
+                    numClasses,
+                    device);
+            }
+
+            public ResNet(string name, Func<int, int, int, Module> block, int expansion, IList<int> num_blocks, int numClasses, Device device = null) : base(name)
+            {
+                if (planes.Length != strides.Length) throw new ArgumentException("'planes' and 'strides' must have the same length.");
+
+                var modules = new List<(string, Module)>();
+
+                conv1 = Conv2d(3, 64, kernelSize: 7, stride: 2, padding: 3, bias: false);
+                bn1 = BatchNorm2d(64);
+                relu1 = ReLU(inPlace: true);
+                maxpool = MaxPool2d(kernelSize: 2, stride: 2, padding: 1);
+                MakeLayer(layer1, block, expansion, 64, num_blocks[0], 1);
+                MakeLayer(layer2, block, expansion, 128, num_blocks[1], 2);
+                MakeLayer(layer3, block, expansion, 256, num_blocks[2], 2);
+                MakeLayer(layer4, block, expansion, 512, num_blocks[3], 2);
+                avgpool = nn.AdaptiveAvgPool2d(new long[] { 1, 1 });
+                flatten = Flatten();
+                fc = Linear(512 * expansion, numClasses);
+
+                RegisterComponents();
+
+                if (device != null && device.type == DeviceType.CUDA)
+                    this.to(device);
+            }
+
+            private void MakeLayer(TorchSharp.Modules.ModuleList modules, Func<int, int, int, Module> block, int expansion, int planes, int num_blocks, int stride)
```

Comment:
>planes [](http://example.com/codeflow?start=128&length=6)

using this name here is kind of confusing with the array defined with the same name above. would be nice if we can distinguish them.